ui/tract-renderer/src/main/java/org/cru/godtools/tract/activity/TractActivity.kt

```diff
@@ -104,6 +103,11 @@ class TractActivity : BaseToolActivity<TractActivityBinding>(R.layout.tract_acti
         // track this view
         if (savedInstanceState == null) dataModel.tool.value?.let { trackToolOpen(it) }
 
+        // track if activity was launched by a shortcut
+        if(shortCutLaunch) {
+            eventBus.post(ToolOpenedViaShortcutAnalyticsActionEvent)
+        }
```

Comment:
let's move the shortcut launch detection logic to the `BaseToolActivity`, that way it will apply to articles shortcuts as well

Reply:
in addition, Android can recreate activities multiple times while it is being used. The way the logic is currently written this event will fire every time the activity is re-created.

the standard work around pattern we use for that is checking to see if there is `savedInstanceState` present or not

Reply:
Okay I went ahead and made a change so its:
```
 if(shortCutLaunch && savedInstanceState == null) {
            eventBus.post(ToolOpenedViaShortcutAnalyticsActionEvent)
}
```
should that do the trick?

Reply:
yeah, that works good.

A micro optimization is to reverse the order of the 2 conditionals, checking if `savedInstanceState` is null is quicker than loading a value out of the intent to see if it's true or false. boolean operations short-circuit when possible, so it would see that `savedInstanceState` isn't null, at which point it wouldn't even bother checking the intent for the flag.

Reply:
Makes sense! I'll switch the two.